### PR TITLE
Avoid duplicate request body in gRPC request object

### PR DESCRIPTION
**Pull request by matiasinsaurralde**
Potential fix for #2208, described [here](https://github.com/TykTechnologies/tyk/issues/2208#issuecomment-483262129).
Need to add tests.

### CODE REVIEW

File: coprocess.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func (c *CoProcessor) ObjectFromRequest(r *http.Request) *coprocess.Object {
 	if r.Body != nil {
 		defer r.Body.Close()
 		miniRequestObject.RawBody, _ = ioutil.ReadAll(r.Body)
```
Comment (dencoded) [inline coprocess.go]: worth to catch and log error here

Comment (matiasinsaurralde) [reply coprocess.go]: Have extended `ObjectFromRequest` to also return errors 👍

**Comment by matiasinsaurralde** [comment]
Fixed all the issues, the failing test seems to be`TestRateLimitForAPIAndRateLimitAndQuotaCheck` (not related)

**Comment by matiasinsaurralde** [comment]
A note about this, when reconstructing the request object after the hook is called, we always take `raw_body`:
```go
// ObjectPostProcess does CoProcessObject post-processing (adding/removing headers or params, etc.).
func (c *CoProcessor) ObjectPostProcess(object *coprocess.Object, r *http.Request) {
        r.ContentLength = int64(len(object.Request.RawBody))
        r.Body = ioutil.NopCloser(bytes.NewReader(object.Request.RawBody))
```
I don't think that this behavior should be modified when`raw_body_only` flag is in use.
We should clarify in the docs that when a hook needs to modify the body, they should modify `raw_body`, perhaps we can mention that the `body` field is "read-only" (we don't have a way of enforcing this, as the bindings are generated by PB, but any hook that modifies `body` won't make any effect, `raw_body` will still be used).

**Comment by matiasinsaurralde** [comment]
Would appreciate any comments on this ☝️ 
It's mostly about the docs.

**Comment by matiasinsaurralde** [comment]
Fixed to match the new structures
Tests ok @dencoded 👍

**Comment by dencoded** [comment]
👍 to go then

**Comment by buger** [comment]
@matiasinsaurralde not quite sure how it works. How I can set `RawBodyOnly` option?  Can you provide an example?

**Comment by matiasinsaurralde** [comment]
@buger this seems broken after the rebase, I will add additional tweaks and clarify the functionality

**Pull request merged**